src/utils/flatten.js

```diff
@@ -52,6 +53,20 @@ const flatten = (
       return ruleSet
     }
 
+    /* Throw if a React Element was given styles */
+    if (React.isValidElement(chunk)) {
```

Comment:
Hmm this seems to be happening too late. We should be doing this check before the interpolated function is executed so we can get the assigned name of the actual component class.

For example:

```js
const NestedComp = () => <button />
const Thing = styled.div`
  ${NestedComp} {
    color: blue;
  }
`
```

Should result in the error being like this:

A plain React class (NestedComp) was interpolated in your styles [...]